giotto/homology/grids.py

```diff
@@ -0,0 +1,182 @@
+        values within all images passed to meth:`fit`.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    See also
+    --------
+    VietorisRipsPersistence
+
+    Notes
+    -----
+    `Gudhi <https://github.com/GUDHI/gudhi-devel>`_ is used as a C++ backend
+    for computing Cubical persistent homology. Python bindings were modified
+    for performance.
+
+    Persistence diagrams produced by this class must be interpreted with
+    care due to the presence of padding triples which carry no information.
+    See :meth:`transform` for additional information.
+
+    References
+    ----------
+    [1] P. Dlotko, "Cubical complex", 2015; `GUDHI User and Reference Manual \
+    <http://gudhi.gforge.inria.fr/doc/latest/group__cubical__complex.html>`_.
+
+    """
+    _hyperparameters = {'infinity_values_': [numbers.Number],
+                        '_homology_dimensions': [list, [int, (0, np.inf)]],
+                        'coeff': [int, (2, np.inf)]}
+
+    def __init__(self, homology_dimensions=(0, 1), coeff=2,
+                 infinity_values=None, n_jobs=None):
+        self.homology_dimensions = homology_dimensions
+        self.coeff = coeff
+        self.infinity_values = infinity_values
+        self.n_jobs = n_jobs
+
+    def _gudhi_diagram(self, X):
+        cubical_complex = CubicalComplex(
+            dimensions=X.shape,
+            top_dimensional_cells=X.flatten(order="F"))
+        Xds = cubical_complex.persistence(homology_coeff_field=self.coeff,
+                                          min_persistence=0)
+        Xds = {dim: np.array([Xds[i][1] for i in range(len(Xds))
+                              if Xds[i][0] == dim]).reshape((-1, 2))
+               for dim in self.homology_dimensions}
+        Xds = {dim: np.hstack([Xds[dim], dim * np.ones((Xds[dim].shape[0], 1),
+                                                       dtype=Xds[dim].dtype)])
+               for dim in self._homology_dimensions}
+        return Xds
+
+    def fit(self, X, y=None):
+        """Do nothing and return the estimator unchanged.
+
+        This method is there to implement the usual scikit-learn API and hence
+        work in pipelines.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_pixels_1, ..., n_pixels_d)
+            Input data. Array of d-dimensional images.
+
+        y : None
+            There is no need of a target in a transformer, yet the pipeline API
+            requires this parameter.
+
+        Returns
+        -------
+        self : object
+
+        """
+        if self.infinity_values is None:
+            self.infinity_values_ = np.max(X)
+        else:
+            self.infinity_values_ = self.infinity_values
+
+        self._homology_dimensions = sorted(self.homology_dimensions)
```

Comment:
Would `sorted_homology_dimensions` be clearer instead of `_homology_dimensions`?

Reply:
It should be an attribute so `_sorted_homology_dimensions`. But all over the code `_homology_dimensions` is being used for the same thing, so I would keep it for consistency.